### PR TITLE
Hamza week2

### PR DESCRIPTION
**Pull request by lizozo**
Sorry for the mess

week-2 files:
1.world.js
2.checkNumberOfLAnguages.js
3.ER-todo.png

### CODE REVIEW

File: Week2/checkNumberOfLAnguages.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+}
+
+let queries = [
+  'CREATE DATABASE IF NOT EXISTS test_world',
+  'USE test_world',
+  'DROP TABLE IF EXISTS countrylanguage',
+
+  'CREATE TABLE countrylanguage (name VARCHAR(56), language VARCHAR(50))',
+
+  'CREATE TRIGGER `countrylanguage_AFTER_INSERT` AFTER INSERT ON `countrylanguage` FOR EACH ROW BEGIN ' +
```
Comment (gajduk) [inline Week2/checkNumberOfLAnguages.js]: nice 👍

File: Week2/world.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+    type: 'string',
+    required: true,
+    ask: function() {
+      return prompt.history('query').value == 3 || prompt.history('query').value == 4;
+    },
+  },
+];
+
+function createJoin(column1, column2, table1, table2, onColumn1, onColumn2, value1, value2) {
+  let sql = ' select distinct ??, ?? from ?? left join ?? on ?? = ?? where ?? = ?';
```
Comment (gajduk) [inline Week2/world.js]: using double question marks to escape identifiers, very good 👍

File: Week2/world.js [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  return selection;
+}
+
+async function main() {
+  try {
+    const { query, country, region, language } = await getPromptInputs();
+    switch (query) {
+      case '1':
+        const capital = await execQuery(
+          await createJoin(
```
Comment (gajduk) [inline Week2/world.js]: this is a really small thing but createJoin is not an async function so there is no need to await for it

File: Week2/world.js [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+function createQueryFour(language, region) {
+  let sql =
+    "select country.Name, countrylanguage.language, countrylanguage.isOfficial from countrylanguage left join country on country.Code =countrylanguage.CountryCode where country.Region = ? and countrylanguage.language = ? and countrylanguage.isofficial = 'T'";
+  const inserts = [region, language];
+  sql = mysql.format(sql, inserts);
+  return sql;
+}
+function createQueryFive() {
+  let sql =
+    'select country.continent, count(distinct(countrylanguage.language)) as count from country left join countrylanguage on country.code = countrylanguage.countryCode group by country.continent';
```
Comment (gajduk) [inline Week2/world.js]: used distinct 👍

**Comment by lizozo** [comment]
Hi @gajduk !! thank you very much four your time and comments. I really appreciate your efforts that you do for us.

**Pull request closed without merging**